src/common/Cryptography/SHA1.h

```diff
@@ -29,21 +30,34 @@ class TC_COMMON_API SHA1Hash
     public:
         SHA1Hash();
         ~SHA1Hash();
+        static constexpr size_t HASH_LEN = SHA_DIGEST_LENGTH;
 
-        void UpdateBigNumbers(BigNumber* bn0, ...);
 
         void UpdateData(const uint8 *dta, int len);
         void UpdateData(const std::string &str);
 
+        template <size_t... SIZES, typename... Ts> auto UpdateBigNumbers(Ts&&... ts) -> std::enable_if_t<(std::is_same_v<BigNumber, std::decay_t<Ts>>&& ...)>
+        {
+            static_assert(sizeof...(SIZES) == sizeof...(Ts), "Size count needs to equal argument count");
+            (UpdateBigNumber<SIZES>(std::forward<Ts>(ts)), ...);
+        }
+
         void Initialize();
         void Finalize();
 
-        uint8 *GetDigest(void) { return mDigest; }
-        int GetLength(void) const { return SHA_DIGEST_LENGTH; }
+        uint8 const* GetDigest(void) const { return _mDigest; }
 
     private:
-        SHA_CTX mC;
-        uint8 mDigest[SHA_DIGEST_LENGTH];
+        static void _CheckedBNToBytes(BigNumber const& bn, uint8* arena, size_t sz);
+        template <size_t SIZE> void UpdateBigNumber(BigNumber const& bn)
+        {
+            std::array<uint8, SIZE> arena;
+            _CheckedBNToBytes(bn, arena.data(), SIZE);
+            UpdateData(arena.data(), SIZE);
+        }
+
+        SHA_CTX _mC;
```

Comment:
thats a not properly executed codestyle adjustment, old name `mC` was a variation of `m_C` without the underscore, you should have removed the m as well